conan/tools/cmake/cmakedeps/cmakedeps.py

```diff
@@ -41,13 +41,16 @@ def content(self):
         build_req = self._conanfile.dependencies.build_requires_build_context
 
         # Check if the same package is at host and build and the same time
-        common = {r.ref.name for r in host_req}.intersection({r.ref.name for r in build_req})
```

Comment:
I think it would be more readable and short just filtering the activated, something like:

```
activated_br = {r.ref.name for r in build_req if r.ref.name in self.build_context_activated}
common_names = {r.ref.name for r in host_req}.intersection(activated_br)
```

Reply:
Changed, please review.

In any case, the important feedback here is if this makes sense as default behavior, or I am missing some other thing.

Reply:
This check? Yes, it was missing and makes perfect sense.